pkg/clusteridentity/clusteridentity.go

```diff
@@ -0,0 +1,175 @@
+)
+
+// ClusterIdentityAllocator ensures that the antrea-cluster-identity ConfigMap is populated
+// correctly, with a valid UUID. It is meant to be used by the Antrea Controller.
+type ClusterIdentityAllocator struct {
+	clusterIdentityConfigMapNamespace string
+	clusterIdentityConfigMapName      string
+	k8sClient                         clientset.Interface
+}
+
+// NewClusterIdentityAllocator creates a ClusterIdentityAllocator object
+func NewClusterIdentityAllocator(
+	clusterIdentityConfigMapNamespace string,
+	clusterIdentityConfigMapName string,
+	k8sClient clientset.Interface,
+) *ClusterIdentityAllocator {
+	return &ClusterIdentityAllocator{
+		clusterIdentityConfigMapNamespace: clusterIdentityConfigMapNamespace,
+		clusterIdentityConfigMapName:      clusterIdentityConfigMapName,
+		k8sClient:                         k8sClient,
+	}
+}
+
+func (a *ClusterIdentityAllocator) updateConfigMapIfNeeded() error {
+	configMap, err := a.k8sClient.CoreV1().ConfigMaps(a.clusterIdentityConfigMapNamespace).Get(context.TODO(), a.clusterIdentityConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error when getting '%s/%s' ConfigMap: %v", a.clusterIdentityConfigMapNamespace, a.clusterIdentityConfigMapName, err)
+	}
+
+	// returns a triplet consisting of the cluster UUID, a boolean indicating if the UUID needs
+	// to be written to the ConfigMap, and an error if applicable
+	inspectUUID := func() (uuid.UUID, bool, error) {
+		clusterUUIDStr, ok := configMap.Data[uuidConfigMapKey]
+		if ok && clusterUUIDStr != "" {
+			clusterUUID, err := uuid.Parse(clusterUUIDStr)
+			if err != nil {
+				return uuid.Nil, false, fmt.Errorf("cluster already has UUID '%s' but it is not valid: %v", clusterUUIDStr, err)
+			}
+			return clusterUUID, false, nil
+		}
+
+		// generate a new random UUID
+		clusterUUID := uuid.New()
+
+		return clusterUUID, true, nil
+	}
+
+	clusterUUID, clusterUUIDNeedsUpdate, err := inspectUUID()
+	if err != nil {
+		return err
+	}
+	if !clusterUUIDNeedsUpdate {
+		klog.Infof("Existing cluster UUID: %v", clusterUUID)
+		return nil
+	}
+
+	configMap.Data = map[string]string{
+		uuidConfigMapKey: clusterUUID.String(),
+	}
+	if _, err := a.k8sClient.CoreV1().ConfigMaps(a.clusterIdentityConfigMapNamespace).Update(context.TODO(), configMap, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("error when updating '%s/%s' ConfigMap with new cluster identity: %v", a.clusterIdentityConfigMapNamespace, a.clusterIdentityConfigMapName, err)
+	}
+	klog.Infof("New cluster UUID: %v", clusterUUID)
+	return nil
+}
+
+// Run will ensure that the antrea-cluster-identity ConfigMap is up-to-date. It is meant to be
+// called asynchronously in its own goroutine, and will keep retrying in case of error, using an
+// exponential backoff mechanism.
+func (a *ClusterIdentityAllocator) Run(stopCh <-chan struct{}) {
+	// exponential backoff, starting at 100ms with a factor of 2. A "steps" value of 10 means we
```

Comment:
Probably capitalize the first letter: Exponential

Reply:
Should we say "value of 8" to match your Backoff struct?

Reply:
Thanks, done. I must have experimented with different values when I was writing the PR.